src/commonMain/kotlin/exchange.dydx.abacus/state/v2/manager/AsyncAbacusStateManagerV2.kt

```diff
@@ -282,27 +282,35 @@ class AsyncAbacusStateManagerV2(
     }
 
     private fun load(configFile: ConfigFile) {
-        val path = configFile.path
-        if (appConfigs.loadRemote) {
-            loadFromRemoteConfigFile(configFile)
-            val configFileUrl = "$deploymentUri$path"
-            ioImplementations.rest?.get(configFileUrl, null, callback = { response, httpCode, headers ->
-                if (success(httpCode) && response != null) {
-                    if (parse(response, configFile)) {
-                        writeToLocalFile(response, path)
-                    }
-                }
-            })
-        } else {
-            loadFromBundledLocalConfigFile(configFile)
+        ioImplementations.threading?.async(ThreadingType.network) {
+            val path = configFile.path
+            if (appConfigs.loadRemote) {
+                loadFromRemoteConfigFile(configFile)
+                val configFileUrl = "$deploymentUri$path"
+                ioImplementations.rest?.get(
+                    configFileUrl,
+                    null,
+                    callback = { response, httpCode, headers ->
+                        if (success(httpCode) && response != null) {
+                            if (parse(response, configFile)) {
+                                writeToLocalFile(response, path)
+                            }
+                        }
+                    },
+                )
+            } else {
+                loadFromBundledLocalConfigFile(configFile)
+            }
         }
     }
 
     private fun loadFromRemoteConfigFile(configFile: ConfigFile) {
         ioImplementations.fileSystem?.readCachedTextFile(
             configFile.path,
         )?.let {
-            parse(it, configFile)
+            ioImplementations.threading?.async(ThreadingType.abacus) {
```

Comment:
i feel like this is a good example of how threading is making things hard to reason about. when we call `loadFromRemoteConfigFile`, we are actually on the main thread since this is called at object initialization time. So shouldn't we be going back to that thread instead of the `abacus` thread?

Reply:
Internal processing is done on the abacus thread.  But I missed the post-processing after the rest call.

Making readCachedTextFile main-safe would require a breaking changing to the protocol to use callbacks.  For now I think we can live with what we have.  Yes, it does have some threading complexity in Abacus, but the touchpoints are rather limited.  Once we have those working properly, it shouldn't require much future maintenance.